internal/characters/thoma/abil.go

```diff
@@ -0,0 +1,237 @@
+	}
+	c.QueueParticle("thoma", count, core.Pyro, f+100)
+
+	shieldamt := (shieldpp[c.TalentLvlSkill()]*c.MaxHP() + shieldflat[c.TalentLvlSkill()])
+	c.genShield("Thoma Skill", shieldamt)
+
+	// damage component not final
+	x, y := c.Core.Targets[0].Shape().Pos()
+	c.Core.Combat.QueueAttack(ai, core.NewCircleHit(x, y, 2, false, core.TargettableEnemy), f+1, f+1)
+	
+	cd := 15
+	if c.Base.Cons >= 1 {
+		cd = 12; //the CD reduction activates when a character protected by Thoma's shield is hit. Since it is almost impossible for this not to activate, we set the duration to 12 for sim purposes.
+	}
+	c.SetCD(core.ActionSkill, cd*60)
+	return f, a
+}
+
+// Burst attack damage queue generator
+func (c *char) Burst(p map[string]int) (int, int) {
+
+	f, a := c.ActionFrames(core.ActionBurst, p)
+
+	ai := core.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Crimson Ooyoroi",
+		AttackTag:  core.AttackTagElementalBurst,
+		ICDTag:     core.ICDTagNone,
+		ICDGroup:   core.ICDGroupDefault,
+		StrikeType: core.StrikeTypeDefault,
+		Element:    core.Pyro,
+		Durability: 50,
+		Mult:       burst[c.TalentLvlSkill()],
+	}
+
+	// damage component not final
+	x, y := c.Core.Targets[0].Shape().Pos()
+	c.Core.Combat.QueueAttack(ai, core.NewCircleHit(x, y, 2, false, core.TargettableEnemy), f+1, f+1)
+	
+	d := 15
+	if c.Base.Cons >= 2 {
+		d = 18
+	}
+
+	c.Core.Status.AddStatus("thomaburst", d*60)
+
+	c.burstProc()
+	
+	if c.Base.Cons >= 4 {
+		c.AddTask(func() {
+			c.c4Restore()
+		}, "thoma-c4-restore", 15)
+	}
+
+	cd := 20
+	if c.Base.Cons >= 1 {
+		cd = 17; //the CD reduction activates when a character protected by Thoma's shield is hit. Since it is almost impossible for this not to activate, we set the duration to 17 for sim purposes.
+	}
+	c.SetCDWithDelay(core.ActionBurst, cd*60, 11)
+	c.ConsumeEnergy(11)
+
+	return f, a
+}
+
+func (c *char) burstProc() {
+	// does not deactivate on death
+	icd := 0
+	c.Core.Events.Subscribe(core.OnDamage, func(args ...interface{}) bool {
+		ae := args[1].(*core.AttackEvent)
+		t := args[0].(core.Target)
+		if ae.Info.AttackTag != core.AttackTagNormal && ae.Info.AttackTag != core.AttackTagExtra {
+			return false
+		}
+		if c.Core.Status.Duration("thomaburst") == 0 {
+			return false
+		}
+		if icd > c.Core.F {
+			c.Core.Log.NewEvent("thoma Q (active) on icd", core.LogCharacterEvent, c.Index, "frame", c.Core.F)
+			return false
+		}
+
+		ai := core.AttackInfo{
+			ActorIndex: c.Index,
+			Abil:       "Fiery Collapse",
+			AttackTag:  core.AttackTagElementalBurst,
+			ICDTag:     core.ICDTagElementalBurst,
+			ICDGroup:   core.ICDGroupDefault,
+			StrikeType: core.StrikeTypeDefault,
+			Element:    core.Pyro,
+			Durability: 50,
+			Mult:       burstproc[c.TalentLvlSkill()],
+			FlatDmg:    0.022 * c.HPMax,
+		}
+		//trigger a chain of attacks starting at the first target
+		atk := core.AttackEvent{
+			Info: ai,
+		}
+		atk.SourceFrame = c.Core.F
+		atk.Pattern = core.NewDefSingleTarget(t.Index(), core.TargettableEnemy)
```

Comment:
@HanSeoulOh @kurt22i Nitpick here, so far as I can tell, "Fiery Collapse" is an AOE that fires in the direction of the attack of the active character. Here's a quick recording I took and we can see the collapse hitting both the abyss mage and the hilichurl blockade: https://streamable.com/pznwrw

Reply:
yeah i think i never implemented the AOE cause wasn't sure what to do with it @srliao any suggestions?